### ISSUE TITLE
Add ELI5 long form QA dataset

### ISSUE THREAD

**Issue by yjernite** [facebookresearch/ParlAI#2094]
**Is your feature request related to a problem? Please describe.**
The ELI5 task requires a system to read a question and support document made up of passages from Common Crawl web pages, and write a multi-sentence answer, as described in https://arxiv.org/abs/1907.09190. The questions and answers are taken from the ELI5 subreddit.

The dataset is provided in the form of a mapping from ELI5 questions to Common Crawl IDs for relevant pages and a set of scripts to download and filter Reddit data from https://pushshift.io and Common Crawl data from AWS: https://github.com/facebookresearch/ELI5.

The process is pretty involved, requires several steps to be run over a period of days, and requires access to a large SLURM cluster. We would like to simplify it so that users can more easily re-create the dataset and use it within ParlAI.

**Describe the solution you'd like**
Write general purpose scripts to download Reddit dumps from pushshift.io and gather all posts from specific sub-reddits or with specific IDs

Write scripts to gather pages from the WET Common Crawl dumps by URL or CCUID

Write single wrapper script to download the Reddit data, download the Common Crawl pages, select passages to create support document, and split dataset into train, valid, and test.

**Additional context**
Many of the required functions are generally useful for any dataset built from Reddit or the Common Crawl, the API should be general purpose and modular

A lot of the time is spent downloading Common Crawl slices from S3. Consider using AWS for some of the processing to speed up that part.

**Comment by stephenroller** [comment]
We have Jason's internal teacher. It's not unreasonable to just make that public and have it look for the files produced by the eli5 scripts and tell you "put them in this folder."

**Comment by stephenroller** [comment]
Fixed in #2467
